### PR TITLE
Task-51370: Analytics to know the use of unified search filtering

### PR DESCRIPTION
**Pull request by AzmiTouil**
Add statistics when user use the filter in the unified search

### CODE REVIEW

File: component/core/src/main/java/org/exoplatform/social/core/processor/MentionsProcessor.java [hunk 1]
```diff
@@ -62,6 +62,9 @@ public void processActivity(ExoSocialActivity activity) {
       if (templateParams.containsKey("default_title")) {
         templateParams.put("default_title", MentionUtils.substituteUsernames(portalOwner, templateParams.get("default_title")));
       }
+      if (templateParams.containsKey("kudosMessage")) {
+        templateParams.put("kudosMessage", MentionUtils.substituteUsernames(portalOwner, templateParams.get("kudosMessage")));
```
Comment (boubaker) [inline component/core/src/main/java/org/exoplatform/social/core/processor/MentionsProcessor.java]: plz no Kudos in Social layer

File: webapp/portlet/src/main/webapp/vue-apps/search/components/SearchResults.vue [hunk 2]
```diff
@@ -420,6 +422,41 @@ export default {
           });
       });
     },
+    addSearchFilterAnalytics(connectorName) {
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/search/components/SearchResults.vue]: plz no Analytics in Social layer

File: webapp/portlet/src/main/webapp/vue-apps/search/components/SearchResults.vue [hunk 3]
```diff
@@ -290,16 +290,39 @@ export default {
...
           'subModule': 'ui',
           'userId': eXo.env.portal.userIdentityId,
           'userName': eXo.env.portal.userName,
           'operation': 'click',
-          'name': `search${selectedConnector.name.charAt(0).toUpperCase()}${selectedConnector.name.slice(1)}`,
+          'name': name,
           'timestamp': Date.now()
         };
-        document.dispatchEvent(new CustomEvent('exo-statistic-message', {detail: searchFilterAnalytics}));
+        document.dispatchEvent(new CustomEvent('exo-statistic-message', {detail: searchFilter}));
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/search/components/SearchResults.vue]: We can't add statistics collection here, all you can do is to trigger a generic event that should be catched by eXo Analytics addon to collect statistics (No dependency from social to Analytics should be visible in source code of Meeds).

```suggestion
        document.dispatchEvent(new CustomEvent('search-filter-selected', {detail: selectedConnector.name}));
```

And we should delete all the upper computation that builds an analytics object `searchFilter` which should be made in Analytics addon only.

Comment (AzmiTouil) [reply webapp/portlet/src/main/webapp/vue-apps/search/components/SearchResults.vue]: Updated !

**Review by nidhalnidhal — APPROVED** [review]

**Pull request merged**